autosynth/multi.py

```diff
@@ -16,83 +18,64 @@
-            args,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-            cwd=cwd,
-            check=check,
-            encoding="utf-8",
-            env=env,
-        )
-    except subprocess.CalledProcessError as exc:
-        logger.error(
-            f"Failed executing {' '.join((str(arg) for arg in args))}:\n\n"
-            "{exc.stdout}"
-        )
-        raise exc
+def synthesize_library(
+    library: typing.Dict, github_token: str, extra_args: typing.List[str]
+) -> typing.Dict:
+    """Run autosynth on a single library.
 
+    Arguments:
+        library {dict} - Library configuration
 
-def synthesize_libraries(libraries, github_token, extra_args):
-    results = []
-    for library in libraries:
-        logger.info(f"Synthesizing {library['name']}.")
-
-        command = [sys.executable, "-m", "autosynth.synth"]
-
-        env = os.environ
-        env["GITHUB_TOKEN"] = github_token
-
-        library_args = [
-            "--repository",
-            library["repository"],
-            "--synth-path",
-            library.get("synth-path", ""),
-            "--branch-suffix",
-            library.get("branch-suffix", ""),
-            "--pr-title",
-            library.get("pr-title", ""),
-        ]
-
-        if library.get("metadata-path"):
-            library_args.extend(["--metadata-path", library.get("metadata-path")])
-
-        if library.get("deprecated-execution", False):
-            library_args.append("--deprecated-execution")
-
-        result = run(
-            command + library_args + library.get("args", []) + extra_args,
-            check=False,
-            env=env,
-        )
+    """
+    logger.info(f"Synthesizing {library['name']}.")
 
-        results.append({"config": library, "result": result})
+    command = [sys.executable, "-m", "autosynth.synth"]
 
-    return results
+    env = os.environ
+    env["GITHUB_TOKEN"] = github_token
 
+    library_args = [
+        "--repository",
+        library["repository"],
+        "--synth-path",
+        library.get("synth-path", ""),
+        "--branch-suffix",
+        library.get("branch-suffix", ""),
+        "--pr-title",
+        library.get("pr-title", ""),
+    ]
 
-def _format_result(result):
-    error = False
-    skipped = False
-    returncode = result["result"].returncode
+    if library.get("metadata-path"):
+        library_args.extend(["--metadata-path", library.get("metadata-path")])
 
-    if returncode != 0:
-        if returncode == 28:
-            skipped = True
-        else:
-            error = True
+    if library.get("deprecated-execution", False):
+        library_args.append("--deprecated-execution")
 
+    result = executor.run(
+        command + library_args + library.get("args", []) + extra_args,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        check=False,
+        encoding="utf-8",
+        env=env,
+    )
+    error = result.returncode not in (0, 28)
```

Comment:
Now that synthtool and autosynth live in the same repo, we can put the magic number 28 into a constant and both code bases can import it.